src/test/java/organice/logic/commands/MatchCommandTest.java

```diff
@@ -0,0 +1,351 @@
+        String validNric = VALID_NRIC_PATIENT_IRENE;
+
+        CommandResult commandResult = new MatchCommand(validNric).execute(modelStub);
+        ObservableList<Person> listOfDonors = modelStub.getFilteredPersonList();
+
+        assertEquals(String.format(MatchCommand.MESSAGE_SUCCESS, validNric), commandResult.getFeedbackToUser());
+        assertEquals(Arrays.asList(DONOR_IRENE_DONOR), listOfDonors);
+    }
+
+    @Test
+    public void execute_patientExists_noMatch() throws Exception {
+        ModelStubWithoutDonor modelStub = new ModelStubWithoutDonor(PATIENT_IRENE);
+        String validNric = VALID_NRIC_PATIENT_IRENE;
+
+        CommandResult commandResult = new MatchCommand(validNric).execute(modelStub);
+        ObservableList<Person> listOfDonors = modelStub.getFilteredPersonList();
+
+        assertEquals(String.format(MatchCommand.MESSAGE_SUCCESS, validNric), commandResult.getFeedbackToUser());
+        assertEquals(listOfDonors.size(), 0); //assert that there is no donor
+    }
+
+    @Test
+    public void execute_patientDoesNotExist_throwsCommandException() throws Exception {
+        ModelStubWithoutDonor modelStub = new ModelStubWithoutDonor(PATIENT_BOB);
+
+        String validNric = VALID_NRIC_PATIENT_IRENE;
+
+        CommandResult commandResult = new MatchCommand(validNric).execute(modelStub);
+        assertEquals(String.format(MatchCommand.MESSAGE_PERSON_NOT_FOUND, validNric),
+                commandResult.getFeedbackToUser());
+    }
+
+    @Test
+    public void match_isSuccessfulMatch() throws Exception {
+        MatchCommand matchCommand = new MatchCommand(VALID_NRIC_PATIENT_IRENE);
+
+        boolean matchResult = matchCommand.match(DONOR_IRENE_DONOR, PATIENT_IRENE);
+        assertTrue(matchResult); //100% match and same blood type
+
+        //donor with 4/6 matching tissues and same blood type.
+        matchResult = matchCommand.match(DONOR_SIMILAR_TISSUE, PATIENT_IRENE);
+        assertTrue(matchResult);
+
+        //donor with blood type A matches patient with blood type A and AB
+        assertEquals(matchCommand.match(DONOR_A, PATIENT_A), true);
+        assertEquals(matchCommand.match(DONOR_A, PATIENT_AB), true);
+
+        //donor with blood type B matches patient with blood type B and AB
+        assertEquals(matchCommand.match(DONOR_B, PATIENT_B), true);
+        assertEquals(matchCommand.match(DONOR_B, PATIENT_AB), true);
+
+        //donor with blood type AB matches patient with blood type AB
+        assertEquals(matchCommand.match(DONOR_AB, PATIENT_AB), true);
+
+        //donor with blood type O matches patient with blood type A, B, AB and O
+        assertEquals(matchCommand.match(DONOR_O, PATIENT_A), true);
+        assertEquals(matchCommand.match(DONOR_O, PATIENT_B), true);
+        assertEquals(matchCommand.match(DONOR_O, PATIENT_AB), true);
+        assertEquals(matchCommand.match(DONOR_O, PATIENT_O), true);
```

Comment:
Should we have some assertfalse test whereby the bloodtype should return unmatch?

Reply:
The tests are in the `match_isNotSuccessfulMatch()` method